fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -198,13 +203,26 @@ def predict(self, X):
     def _pmf_predict(self, X):
         """Probability mass function for the given input data.
 
+        Calculates predictions for each of the underlying classifiers. The predictions are
+        subsequently multiplied by the classifiers' weights and aggregated into the returned
+        probabilities.
+
         :param X: Feature data
         :type X: numpy.ndarray or pandas.DataFrame
         :return: Array of tuples with the probabilities of predicting 0 and 1.
         :rtype: pandas.DataFrame
         """
-        pred = pd.DataFrame()
-        for t in range(len(self._hs)):
-            pred[t] = self._hs[t](X)
+        n_samples = len(X)
+
+        def get_weighted_predictions(row):
```

Comment:
rename: it's not a weighted prediction, weighting happens later

Reply:
I think that this was implemented in #442 (slightly more transparently)--but the docstring for `_pmf_predict` needs to be updated there appropriately.

Reply:
removed this as agreed, and will update #442 with more detail